brokerbot.go

```diff
@@ -104,18 +104,31 @@ func main() {
 		port = "8080"
 	}
 
+	server := &http.Server{
+		Addr: ":" + port,
+	}
+	shutdownlib.AddShutdownHandler((func() error {
+		log.Printf("BrokerBot shutting down HTTP server.")
+		return server.Shutdown(ctx)
+	}))
+
 	log.Printf("BrokerBot ready to serve on port %s", port)
-	if err := http.ListenAndServe(":"+port, nil); err != nil {
-		discordClient.Close()
-		log.Fatal(err)
+	if err := server.ListenAndServe(); err != nil {
+		if err != http.ErrServerClosed {
+			discordClient.Close()
+			log.Fatal(err)
+		}
 	}
+	shutdownlib.WaitForShutdown()
 }
 
 func initTokens() {
 	if *discordToken != "" && *finnhubToken != "" {
 		return
 	}
 
+	log.SetFlags(0) // Disable timestamps when using Cloud Logging.
```

Comment:
Should we condition this on the test flag too? if its in test mode, we might want the timestamps since we don't have Cloud Logging

Reply:
We only use test mode locally, in combination with the token flags; this only disables the timestamps once we've fallen back to Secret Manager which only works from Cloud Run.